### ISSUE TITLE
Node Engine concurrent map write

### ISSUE THREAD

**Issue by giobart** [oakestra/oakestra#265]
## Short
There is a concurrency issue in the current node engine implementation. 

## Proposal 
Stack Trace:
```
INFO-2023/12/14 14:48:28 ContainersManagement.go:140: Sending kill signal to mb.test.bench.test.instance.2                               
fatal error: concurrent map writes                                                                                                       
                                                                                                                                         
goroutine 1021781 [running]:                                                                                                             
go_node_engine/virtualization.(*ContainerRuntime).containerCreationRoutine.func2()                                                       
        /Users/giovannibartolomeo/Documents/Projects/edgeio/go_node_engine/virtualization/ContainersManagement.go:238 +0x190             
go_node_engine/virtualization.(*ContainerRuntime).containerCreationRoutine(0x1268900, {0xdacca8?, 0xc000498150}, {0xdb2290?, 0xc000290630
}, 0xc000478000, 0xc001e3e180, 0xc001e3e240, 0xc00053a000, 0xce13a8)                                                                     
        /Users/giovannibartolomeo/Documents/Projects/edgeio/go_node_engine/virtualization/ContainersManagement.go:305 +0x12df            
created by go_node_engine/virtualization.(*ContainerRuntime).Deploy                                                                      
        /Users/giovannibartolomeo/Documents/Projects/edgeio/go_node_engine/virtualization/ContainersManagement.go:115 +0x8a5             
                                                                                                                                         
goroutine 1 [chan receive, 1378 minutes]:                                                                                                
main.main()                                                                                                                              
        /Users/giovannibartolomeo/Documents/Projects/edgeio/go_node_engine/NodeEngine.go:54 +0x274 

...
...
goroutine 1022184 [select]:
go_node_engine/virtualization.(*ContainerRuntime).containerCreationRoutine(0x1268900, {0xdacca8?, 0xc000498150}, {0xdb2290?, 0xc0004e94a0}, 0xc000478b40, 0xc000194120, 0xc000194180, 0xc0002b0000, 0xce13a8)
        /Users/giovannibartolomeo/Documents/Projects/edgeio/go_node_engine/virtualization/ContainersManagement.go:296 +0x1033
created by go_node_engine/virtualization.(*ContainerRuntime).Deploy
        /Users/giovannibartolomeo/Documents/Projects/edgeio/go_node_engine/virtualization/ContainersManagement.go:115 +0x8a5

goroutine 1022510 [chan receive]:
go_node_engine/virtualization.(*ContainerRuntime).Deploy(0x1268900, 0xc000c84a50, 0xce13a8)
        /Users/giovannibartolomeo/Documents/Projects/edgeio/go_node_engine/virtualization/ContainersManagement.go:126 +0x8c5
go_node_engine/mqtt.deployHandler.func1()
        /Users/giovannibartolomeo/Documents/Projects/edgeio/go_node_engine/mqtt/MqttClient.go:107 +0x8f
created by go_node_engine/mqtt.deployHandler
        /Users/giovannibartolomeo/Documents/Projects/edgeio/go_node_engine/mqtt/MqttClient.go:105 +0x23f
```

How to reproduce: Try scaling to 20/30 instances of the same application, and this issue will likely happen.

## Solution 
Investigating

## Status
Looking for a solution

## Checklist
- [ ] Discussed
- [ ] Solved
- [ ] Tested

**Comment by giobart** [comment]
This looks likely caused by concurrent interactions with containerd sdk. This can be solved together with #264 .

**Comment by giobart** [comment]
Not able to replicate this issue anymore
